src/App.tsx

```diff
@@ -14,11 +14,156 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+interface State {
+  selectedGoods: string[];
+  isActive:boolean;
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+    isActive: false,
+  };
+
+  createMessage = (
+    goods: string[],
+  ) => {
+    const { length } = goods;
+
+    switch (length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${goods[0]} is selected`;
+      case 2:
+        return `${goods.join(' and ')} are selected`;
+      default:
+        return `${goods.slice(0, -1).join(', ')} and ${
+          goods[length - 1]
+        } are selected`;
+    }
+  };
+
+  addGood = (item: string) => {
+    this.setState((prevState) => ({
+      selectedGoods: [...prevState.selectedGoods, item],
+    }));
+
+    this.state.isActive = false;
+  };
+
+  removeGood = (item: string) => {
+    this.setState((prevState) => ({
+      selectedGoods: prevState.selectedGoods.filter((good) => good !== item),
+    }));
+  };
+
+  clear = () => {
+    this.setState({ selectedGoods: [] });
+    this.state.isActive = true;
+  };
+
+  hideClearButton = () => {
+    if (this.state.selectedGoods.length <= 1) {
+      this.state.isActive = true;
+    }
+  };
+
+  render() {
+    const { selectedGoods, isActive } = this.state;
+
+    const showOrHideButton = isActive
+      ? null
+      : (
```

Comment:
can you use logical && and maybe you can use the array length so you dont need `isActive` at all